src/main/java/com/zachary_moore/lint/LintRule.java

```diff
@@ -38,28 +41,45 @@ private LintRule(LintLevel level,
      * @throws LintImplementation.NoReportSetException if there was no report set during runtime of our {@link LintImplementation}'s report
      */
     @SuppressWarnings("unchecked")
-    public Map<JSONFile, List<String>> lint(JSONFile ...jsonFiles) throws LintImplementation.NoReportSetException{
-        Map<JSONFile, List<String>> reportMessages = new HashMap<>();
+    public Map<JSONFile, List<Error>> lint(JSONFile ...jsonFiles) throws LintImplementation.NoReportSetException{
+        Map<JSONFile, List<Error>> reportMessages = new HashMap<>();
+
         for (JSONFile file: jsonFiles) {
 
-            List<?> filteredList = FilterMapper.filter(file, implementation);
+            List<? extends WrappedObject> filteredList = FilterMapper.filter(file, implementation);
             if (filteredList == null) {
                 return null;
             }
 
-            for (Object element : filteredList) {
+            for (WrappedObject element : filteredList) {
                 if (implementation.shouldReport(element)) {
+                    String path = getPath(element);
+
                     if (!reportMessages.containsKey(file)) {
                         reportMessages.put(file, new ArrayList<>());
                     }
-                    reportMessages.get(file).add(implementation.report(element));
+                    reportMessages.get(file).add(new Error(path, implementation.report(element)));
                 }
             }
         }
 
         return reportMessages;
     }
 
+    private String getPath(WrappedObject element) {
```

Comment:
Can this exist as a static util function ? This doesn't seem to need to be in the LintRule Class